### ISSUE TITLE
Enhanced Search results page view

### ISSUE THREAD

**Issue by djwesto** [folklabs/openlocal#81]
Both HH and CPN have asked that:

* the search results page results list can be filtered by category
* category sets be refined into one master category list for use across the entire site

CPN also wanted site-wide cat­egories: We sort all con­tent added to the system into a global set of categories so res­id­ents can quickly dis­cover anything on the site relating to their own area of interest.

I'm wondering if we use something like Solr, could we build this or is it something we can build ourselves perhaps as a View (a la this https://www.lullabot.com/articles/custom-search-forms-with-views-and-fastsearch)

**Comment by djwesto** [comment]
@pmackay you suggested this might be moved to a higher priority since it's core to CPN's spec. Happy to add to a Sprint or add a **p-high** label?

**Comment by pmackay** [comment]
We have site-wide categories and the search results page is faceted, so this feels done to me. Key thing CPN want is separate search for directory.

**Comment by djwesto** [comment]
>We have site-wide categories

Yes, we do have categories and we can hopefully design them with CPN to ensure they're truly _site-wide_ but a category facet doesn't appear in the search result facet list as far as I can tell – those are only to filter by content type, not by category. 

![image](https://cloud.githubusercontent.com/assets/702685/22603507/99314e34-ea3f-11e6-8160-d52a68555e87.png)

However, if we agree to build issue #82 as I've described it, this would render the requirement for site-wide category facets on search result pages less necessary. So it would be worth exploring the possibility of site-wide category facets before dismissing them out of hand.
